Assets/PackageStagingArea/eu.netherlands3d.cameras/Runtime/Scripts/Cameras/Compass.cs

```diff
@@ -0,0 +1,130 @@
+            arrowColor = arrowImage.color;
+
+            //when user rotates, cancel the animation if active
+            cameraActionMap = inputActionAsset.FindActionMap("Camera");
+            rotateAction = cameraActionMap.FindAction("RotateModifier");
+            rotateAction.started += CancelAnimation;
+        }
+
+        private void Update()
+        {
+            arrowTransform.SetRotationZ(cameraTransform.transform.eulerAngles.y);
+            float angle;
+            if(cameraComponent.orthographic)
+            {               
+                angle = Vector3.SignedAngle(cameraTransform.up, Vector3.forward, Vector3.up);                
+            }
+            else
+            {                                
+                angle = Mathf.Rad2Deg * Mathf.Atan2(cameraTransform.forward.z, cameraTransform.forward.x) - 90;                
+            }            
+            if(Mathf.Abs(angle) < northAngleMargin)
+                arrowImage.color = NorthColor;
+            else
+                arrowImage.color = arrowColor;
+        }
+
+        public void SwitchToNorth()
+        {
+            cancelingAnimation = false;
+
+            // Find the tile handlers that are currently active, to disable them at the start of the animation
+            // and reactivate them after the animation to prevent glitching from LOD switches
+            var activeTileHandlers = FindObjectsByType<CartesianTiles.TileHandler>(
+                FindObjectsInactive.Exclude,
+                FindObjectsSortMode.None
+            );
+
+            //set target rotation
+            Quaternion rotateTo = Quaternion.LookRotation(Vector3.forward, Vector3.up);
+            rotateTo.eulerAngles = new Vector3(cameraTransform.eulerAngles.x, rotateTo.eulerAngles.y, cameraTransform.eulerAngles.z);
+
+            StartCoroutine(CameraAnimationLoop(rotateTo, cameraTransform,
+                () => { LockTileHandlers(activeTileHandlers); },
+                () => { cameraTransform.transform.rotation = Quaternion.Slerp(cameraTransform.transform.rotation, rotateTo, Time.deltaTime * snapSpeed); },
+                () => { UnlockTileHandlers(activeTileHandlers); }));
+        }
+
+        private IEnumerator CameraAnimationLoop(Quaternion to, Transform cam, Action onStart, Action onUpdate, Action onEnd)
+        {
+            WaitForUpdate wfu = new WaitForUpdate();
+            onStart.Invoke();
+            while(Quaternion.Angle(cam.rotation, to) > directionMargin && !cancelingAnimation)
+            {
+                onUpdate.Invoke();
+                yield return wfu;
+            }
+            onEnd.Invoke();
+            cancelingAnimation = false;
+        }
+
+        public void CancelAnimation(InputAction.CallbackContext context)
+        {
+            cancelingAnimation = true;
+        }
+
+        private void LockTileHandlers(IEnumerable<CartesianTiles.TileHandler> activeTileHandlers)
```

Comment:
De naam is lichtelijk verwarrend, ik had niet verwacht dat ook het locken van het draggen hierin zat; alleen maar het locken van de tile handlers

Reply:
goeie! Heb het vastzetten van draggen eruit gehaald. Nu zouden alleen de tilehandlers inactief moeten zijn tijdens animaties om spikes te voorkomen. Herrinnert me weer niet zomaar alles te copy pasten :)

Reply:
Oh, het locken was van de drag was een goede juist om te voorkomen dat tijdens de animatie de gebruiker met de camera iets doet, dat kan je animaties in de war schappen. Mijn enige commentaar was de naamgeving :)

Reply:
dan zet ik het weer terug! naam is nu SetTileHandlersEnabled